src/GraphQl/Subscription/SubscriptionIdentifierGenerator.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\GraphQl\Subscription;
+
+/**
+ * Generates an identifier used to identify a subscription.
+ *
+ * @experimental
+ *
+ * @author Alan Poulain <contact@alanpoulain.eu>
+ */
+final class SubscriptionIdentifierGenerator implements SubscriptionIdentifierGeneratorInterface
```

Comment:
Are this class and the related really necessary? Cannot we merge it with the topic generator, and use the topic as ID everywhere?

Reply:
This class is necessary because I use a random string for the subscriptions. I can remove it if I use a hash instead. Maybe using a SHA-256 hash would be a better idea like we discussed since Apollo is doing it for its persisted queries.
Also I cannot add it to the generator since it's not related to Mercure.

Reply:
Using a hash sounds like a good idea!

Reply:
Done. I think keeping the service is a good idea since it allows to change how the identifier is generated if needed.